source/opt/loop_descriptor.h

```diff
@@ -188,6 +242,23 @@ class Loop {
   // This is only to allow LoopDescriptor::dummy_top_loop_ to add top level
   // loops as child.
   friend class LoopDescriptor;
+
+  void FindInductionVariable();
+  bool GetConstant(const ir::Instruction* inst, uint32_t* value) const;
+  bool GetInductionInitValue(const ir::Instruction* variable_inst,
+                             uint32_t* value) const;
+  ir::Instruction* GetInductionStepOperation(
+      const ir::Instruction* variable_inst) const;
+
+  // Returns an OpVariable instruction or null from a load_inst.
+  ir::Instruction* GetVariable(const ir::Instruction* load_inst);
+
+  // Populates the set of basic blocks in the loop.
+  void FindLoopBasicBlocks();
+
+  bool IsLoopInvariant(const ir::Instruction* variable_inst);
+
+  bool IsConstantOnEntryToLoop(const ir::Instruction* variable_inst) const;
```

Comment:
Undefined, not used, and not needed after a LICM pass (which is kind of a basic)